src/charm.py

```diff
@@ -232,18 +228,14 @@ def _get_gateway_apis_kubernetes_resource_manager(self):
     def _get_istioctl(self) -> Istioctl:
         """Return an initialized Istioctl instance."""
         # Default settings
-        setting_overrides = {
-            "components.base.enabled": "true",
-            "components.pilot.enabled": "true",
-        }
+        setting_overrides = {}
 
         # Enable Envoy access logs
         # (see https://istio.io/latest/docs/tasks/observability/logs/access-log/)
         setting_overrides["meshConfig.accessLogFile"] = "/dev/stdout"
 
         # Configure CNI
         # (see https://istio.io/latest/docs/setup/additional-setup/cni/#additional-configuration)
-        setting_overrides["components.cni.enabled"] = "true"
         setting_overrides["values.cni.cniBinDir"] = self.parsed_config["cni-bin-dir"]
         setting_overrides["values.cni.cniConfDir"] = self.parsed_config["cni-conf-dir"]
```

Comment:
Nothing major, but istio started to adopt a [`values.global.platform`](https://istio.io/latest/docs/ambient/install/platform-prerequisites/#microk8s) to select the target k8s distro without specifying target cni locations, I think it doesn't work for all k8s distros yet, but maybe it's smth we want to introduce as well in the future.

Reply:
I think its a great idea to use that.  So much so that there's also #36 :D 

I should have mentioned that in the description.  It felt a little more like a design decision (since its removing charm configs) so I broke it into a separate PR.